tests/protocols/test_http.py

```diff
@@ -1094,6 +1098,19 @@ async def app(scope: Scope, receive: ASGIReceiveCallable, send: ASGISendCallable
     assert not expected_states  # consumed
 
 
+async def test_request_than_limit_max_requests_warn_log(
+    unused_tcp_port: int, http_protocol_cls: HTTPProtocol, caplog: pytest.LogCaptureFixture
+):
+    caplog.set_level(logging.WARNING, logger="uvicorn.error")
+    config = Config(app=app, limit_max_requests=1, port=unused_tcp_port, http=http_protocol_cls)
+    async with run_server(config):
+        async with httpx.AsyncClient() as client:
+            tasks = [client.get(f"http://127.0.0.1:{unused_tcp_port}") for _ in range(3)]
+            responses = await asyncio.gather(*tasks)
+            assert len(responses) == 3
+    assert "Exceeded max request limit ending process." in caplog.text
```

Comment:
This doesn't feel like the right place for this test. We are testing the config/server behavior. Can you find a better place for it, please?

Reply:
Yes, moved to config.